nsqd/client_v2.go

```diff
@@ -214,7 +215,7 @@ func (c *clientV2) Identify(data identifyDataV2) error {
 }
 
 func (c *clientV2) Stats() ClientStats {
-	c.RLock()
+	c.metaLock.RLock()
```

Comment:
FYI this was ultimately the most problematic case.

Previously, this was competing with network operations writing to this client, so when writing blocked, this blocked.  It's called as part of the `/stats` [here](https://github.com/nsqio/nsq/pull/701/files#diff-710982d0c740f14afd7d0153750a9370R140).  Prior to #700, the body of `nsqd.GetStats()` would hold an nsqd-wide lock for the entire lifetime of the function call, meaning that a single blocking client could effectively block any operations that required a top-level lock, or a topic/channel-level lock for whichever the misbehaving client was subscribed to.